lib/watch.js

```diff
@@ -240,12 +243,9 @@ class BrunchWatcher {
 
   restartBrunch(reinstallType) {
     const restart = () => {
-      this.fileList.dispose();
-      this.watcher.close();
-      workers.close();
       // we need this to keep compatibility with global `brunch` binaries
       // from older versions which didn't create a child process
-      if (process.send) {
+      if (process.send && process.env.BRUNCH_FORKED_PROCESS === '1') {
```

Comment:
Added `BRUNCH_FORKED_PROCESS` env to identify when this file is called from `bin/brunch`. Because ava tests run in a subprocess, `process.send('reload');` was being called, but since there is not any handler for reload, the reload never triggered.